storage: use MinLZ compression by default

RaduBerinde opened the pull request:
#### storage: provide option for MinLZ compression

We allow MinLZ on v25.3+ clusters. Note that the v25.2 format supports
MinLZ too (so there are no issues during upgrade).

Epic: none
Release note: None

#### storage: use MinLZ compression by default

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/145401)
<!-- Reviewable:end -->

RaduBerinde commented:
I plan to make the default compression metamorphic (in test builds), but will do it in a separate PR (wanted to make sure everything is green with MinLZ).

sumeerbhola reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/145401)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @jbowens and @xinhaoz)

___
*[`pkg/storage/pebble.go` line 245 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/145401#-OPWkf8F-hlsIQwwx3Yz:-OPWkf8F-hlsIQwwx3Z-:bb0swgq) ([raw file](https://github.com/cockroachdb/cockroach/blob/32a66dcd57eddd59927f9122728a8e0a71ead3d8/pkg/storage/pebble.go#L245)):*
> ```Go
> 	"storage.sstable.compression_algorithm_backup_storage",
> 	`determines the compression algorithm to use when compressing sstable data blocks for backup row data storage;`,
> 	CompressionAlgorithmMinLZ, // Default.
> ```

(drive-by) what is the mixed cluster version story? Presumably in clusters with both v25.1 and v25.2/v25.3 binaries, the former can handle MinLZ sstables (asking since MinLZ wasn't even listed in the enum prior to this change)?


<!-- Sent from Reviewable.io -->

RaduBerinde reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/145401)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @jbowens, @sumeerbhola, and @xinhaoz)

___
*[`pkg/storage/pebble.go` line 245 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/145401#-OPWkf8F-hlsIQwwx3Yz:-OPWmG7ZCQhnlnktnull:bil4wy0) ([raw file](https://github.com/cockroachdb/cockroach/blob/32a66dcd57eddd59927f9122728a8e0a71ead3d8/pkg/storage/pebble.go#L245)):*
<details><summary><i>Previously, sumeerbhola wrote…</i></summary><blockquote>

(drive-by) what is the mixed cluster version story? Presumably in clusters with both v25.1 and v25.2/v25.3 binaries, the former can handle MinLZ sstables (asking since MinLZ wasn't even listed in the enum prior to this change)?
</blockquote></details>

As of https://github.com/cockroachdb/pebble/pull/4648 MinLZ is only supported in the `v6` table format, which is only supported by 25.2+. Older formats fall back to Snappy.


<!-- Sent from Reviewable.io -->

jbowens reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 1 of 1 files at r1, 7 of 7 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/145401#-:-OPX28Jy-pxR_V69zneE:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @sumeerbhola and @xinhaoz)



<!-- Sent from Reviewable.io -->

RaduBerinde reviewed (commented):
TFTRs!

bors r+

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/145401#-:-OPXPMLi4i6mVvRroq9F:b-tlx7ko)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @sumeerbhola and @xinhaoz)

___
*[`pkg/storage/pebble.go` line 245 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/145401#-OPWkf8F-hlsIQwwx3Yz:-OPXPODO8g-ioG9hW9TC:bjnkwee) ([raw file](https://github.com/cockroachdb/cockroach/blob/32a66dcd57eddd59927f9122728a8e0a71ead3d8/pkg/storage/pebble.go#L245)):*
<details><summary><i>Previously, RaduBerinde wrote…</i></summary><blockquote>

As of https://github.com/cockroachdb/pebble/pull/4648 MinLZ is only supported in the `v6` table format, which is only supported by 25.2+. Older formats fall back to Snappy.
</blockquote></details>

It's worth noting that prior to that change, a draft of this PR did fail the upgrade tests.


<!-- Sent from Reviewable.io -->

RaduBerinde commented:
bors r+

RaduBerinde commented:
bors r-

RaduBerinde commented:
Actually, I'm not sure I need that check (though it doesn't hurt). For the cluster to be at v25.3 version, all nodes must have bumped the cluster version to v25.2, which would have ratcheted up to the Pebble format which supports V6.

RaduBerinde committed:
storage: provide option for MinLZ compression

We allow MinLZ on v25.3+ clusters. Note that the v25.2 format supports
MinLZ too (so there are no issues during upgrade).

Epic: none
Release note: None

RaduBerinde committed:
storage: use MinLZ compression by default

RaduBerinde commented:
bors r+
